### PR TITLE
Add support for pathlib for reading PEtab tables

### LINKED ISSUE #82: Add consistent support for pathlib.Path
Would be quite convenient if `Path`s could be used in addition path strings.

### PR DESCRIPTION
**Pull request by dweindl**
Closes #82

### CODE REVIEW

File: petab/yaml.py [hunk 1]
```diff
@@ -178,13 +183,13 @@ def write_yaml(yaml_config: Dict[str, Any], filename: str) -> None:
...
-        observable_files: Union[str, List[str]],
-        yaml_file: str,
-        visualization_files: Optional[Union[str, List[str]]] = None,
+        sbml_files: Union[str, Path, List[str]],
+        condition_files: Union[str, Path, List[str]],
+        measurement_files: Union[str, Path, List[str]],
+        parameter_file: Union[str, Path],
+        observable_files: Union[str, Path, List[str]],
+        yaml_file: Union[str, Path],
+        visualization_files: Optional[Union[str, Path, List[str]]] = None,
```
Comment (dilpath) [inline petab/yaml.py:192]:
```suggestion
        sbml_files: Union[str, Path, List[Union[str, Path]]],
        condition_files: Union[str, Path, List[Union[str, Path]]],
        measurement_files: Union[str, Path, List[Union[str, Path]]],
        parameter_file: Union[str, Path],
        observable_files: Union[str, Path, List[Union[str, Path]]],
        yaml_file: Union[str, Path],
        visualization_files: Optional[Union[str, Path, List[Union[str, Path]]]] = None,
```

**Review by dilpath — APPROVED** [review]
:+1:

Maybe nice to define e.g. `petab.C.TYPE_PATH = Union[str, Path]`, could make the type hints a little neater. Not sure if it can be used as `isinstance(..., TYPE_PATH)` though.

**Comment by dweindl** [comment]
> Maybe nice to define e.g. `petab.C.TYPE_PATH = Union[str, Path]`, could make the type hints a little neater. Not sure if it can be used as `isinstance(..., TYPE_PATH)` though.

Does look very neat, agreed. Briefly thought about it, but then decided that it might be more convenient to users to directly have the types there, without looking up what other type is...

**Comment by dilpath** [comment]
Ah, additionally, the tests were changed to use `Path`, so maybe `str` is no longer tested? Fine as is though

**Pull request merged**